keywrap/keyprovider/keyprovider_test.go

```diff
@@ -0,0 +1,366 @@
+/*
+   Copyright The ocicrypt Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package keyprovider
```

Comment:
Tests look really good! I think adding `err` checking for the calls would be nice to have in case there is weird behavior in an unexpected system/library call that is independent from the feature that causing the tests to misbehave

Reply:
I initially did not add err checking calls, but the golint shows some errors. So I added those